numba/tests/test_num_threads.py

```diff
@@ -0,0 +1,576 @@
+# -*- coding: utf-8 -*-
```

Comment:
We should rename this file, `test_num_threads` doesn't really cover what it's doing. Perhaps `test_thread_masking.py` ?